### DISCUSSION TITLE
Why were the patch versions for CVE-2019-25014, CVE-2022-24726 released so late?

### DISCUSSION THREAD

**Discussion by Silence-worker-02** [istio/istio#47858] [Q&A] [answered]
We are a research team dedicated to Golang, have discovered that CVE-2019-25014, CVE-2022-24726 were addressed in commit d96a9b6ce1ffc3deef46ffd3d356368be228f96a, 40f6caed64e6869a5589f2e2f38f1e64063fd930. However, upon analyzing the commit, we observed that the patch version (1.5.0-alpha.0, 1.14.0-alpha.0) were released after a lapse of over one month. We are interested in understanding the reasons behind this delay in releasing the patch version, as it could potentially impede the prompt dissemination of patches to downstream users. We seek clarification on whether the delay might be attributed to:

1. Issues with testing and CI checking.
2. Other commits requiring inclusion in a single release.
3. By convention, infrequent release of versions.
4. Other reasons.
   We appreciate your attention to this matter and eagerly await your response. Thank you.

**Comment by howardjohn** [comment, marked as answer]
I think you misunderstand. Will focus on CVE-2022-24726. CVE is published in https://istio.io/latest/news/security/istio-security-2022-004/ on Mar 9. The fix is in https://github.com/istio/istio/releases/tag/1.13.2, also published on Mar 9. Same day.


`1.5.0-alpha.0`  and `1.14.0-alpha.0` are ALPHA builds of unreleased versions. They are completely unrelated to security issues, as they are unreleased development only code.

https://github.com/istio/istio/commit/40f6caed64e6869a5589f2e2f38f1e64063fd930 is not related to this CVE at all, its a completely different CVE from Jun 9th: https://istio.io/latest/news/security/istio-security-2022-005/.  The linked commit is the build on `master`, the development build; https://github.com/istio/istio/commit/6ca5055a4db6695ef5504eabdfde3799f2ea91fd is the real one, released in the actual release on Mar 9.
